File: sbschoolmanager-api/src/main/java/com/sbschoolmanager/sbschoolmanager_api/model/Profesor.java
package com.sbschoolmanager.sbschoolmanager_api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;

@Entity
@Table(name ="PROFESOR")
@Getter
@Setter
@NoArgsConstructor
public class Profesor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name =" Cod_Profesor")
    private Integer codProfesor;

    @Column(name= "Nom_CC")
    private String nombre;

    @Column(name= "Ap_profesor")
    private String apellido;

    @Column(name= "DNI_profesor")
    private String dni;

    @Column(name = "Años_experiencia")
    private Integer anosExperiencia;
}

File: sbschoolmanager_app/lib/models/profesor.dart

class Profesor {
  final int? codProfesor;
  final String nombre;
  final String apellido;
  final String dni;
  final int anosExperiencia;

  Profesor({

    this.codProfesor,
    required this.nombre,
    required this.apellido,
    required this.dni,
    required this.anosExperiencia,

  });

  factory Profesor.fromJson(Map<String, dynamic> json){

    return Profesor(
      codProfesor: json["codProfesor"],
      nombre: json["nombre"],
      apellido: json["apellido"],
      dni: json["dni"],
      anosExperiencia: json["anosExperiencia"],
      );
  }

  Map<String, dynamic> toJson(){
    return{
      if(codProfesor != null)
        "codProfesor": codProfesor,
        "nombre": nombre,
        "apellido": apellido,
        "dni": dni,
        "anosExperiencia": anosExperiencia,
    };
  }

}